draft-ietf-jsonpath-base.md

```diff
@@ -881,24 +756,78 @@ Queries:
 
 | Query | Result | Result Paths | Comment |
 | :---: | ------ | :----------: | ------- |
+| `$[*]`   | `{"j": 1, "k": 2}` <br> `[5, 3]` | `$['o']` <br> `$['a']` | Object values      |
 | `$.o[*]` | `1` <br> `2` | `$['o']['j']` <br> `$['o']['k']` | Object values      |
 | `$.o[*]` | `2` <br> `1` | `$['o']['k']` <br> `$['o']['j']` | Alternative result |
 | `$.a[*]` | `5` <br> `3` | `$['a'][0]` <br> `$['a'][1]`     | Array members      |
-{: title="Index wildcard selector examples"}
+{: title="Index wildcard appender examples"}
 
-### Array Slice Selector {#slice}
+### Index selector {#index-selector}
+
+#### Syntax {#syntax-index}
+{: unnumbered}
+
+An index selector `<index>` matches at most one array element value.
+
+~~~~ abnf
+index-selector  = int                             ; decimal integer
+
+int             = ["-"] ( "0" / (DIGIT1 *DIGIT) ) ; -  optional
+DIGIT1          = %x31-39                         ; 1-9 non-zero digit
+~~~~
+
+Applying the numerical `index-selector` is required to select the corresponding
+element. JSONPath allows it to be negative (see {{index-semantics}}).
+
+Notes:
+1. An `index-selector` is an integer (in base 10, as in JSON numbers).
+2. As in JSON numbers, the syntax does not allow octal-like integers with leading zeros such as `01` or `-01`.
+
+#### Semantics {#index-semantics}
+{: unnumbered}
+
+The `index-selector` applied to an array selects an array element using a zero-based index.
+For example, the selector `0` selects the first and the selector `4` the fifth element of a sufficiently long array.
+Nothing is selected, and it is not an error, if the index lies outside the range of the array. Nothing is selected from a value that is not an array.
+
+A negative `index-selector` counts from the array end.
+For example, the selector `-1` selects the last and the selector `-2` selects the penultimate element of an array with at least two elements.
+As with non-negative indexes, it is not an error if such an element does
+not exist; this simply means that no element is selected.
+
+#### Examples
+{: unnumbered}
+
+<!-- EDITING NOTE: There are non-breaking spaces here between j and j -->
+<!-- i.e., j j and not j j -->
+
+JSON:
+
+    {
+      "o": {"j j": {"k.k": 3}},
+      "a": ["a","b"],
+      "'": {"@": 2}
+    }
```

Comment:
The JSON can be stripped back to `["a","b"]` and the examples simplified to `$[1]` and `$[-2]`.